tests/rptest/tests/acls_test.py

```diff
@@ -321,6 +321,91 @@ def test_mtls_principal(self, rules=None, fail=False):
             except ClusterAuthorizationError:
                 assert fail, "list acls should have succeeded"
 
+    @cluster(num_nodes=3)
+    @parametrize(authn_method='sasl')
+    @parametrize(authn_method='mtls_identity')
+    def test_security_feature_migration(self, authn_method: str):
+        def check_perms():
+            # run a few times for good health
+            for _ in range(2):
+                try:
+                    self.get_client("base").acl_list()
+                    self.logger.debug(
+                        'list acls should have failed for base user')
+                    return False
+                except ClusterAuthorizationError:
+                    pass
+
+                # should succeed
+                try:
+                    self.get_client("cluster_describe").acl_list()
+                except Exception:
+                    self.logger.debug(
+                        'list acls should have passed for cluster user')
+                    return False
+
+                try:
+                    self.get_super_client().acl_list()
+                except Exception:
+                    self.logger.debug(
+                        'list acls should have passed for super user')
+                    return False
+
+                return True
+
+        self.prepare_cluster(use_tls=True, use_sasl=True)
+
+        wait_until(check_perms,
+                   timeout_sec=90,
+                   err_msg='check perms failed before migration')
+        '''
+        Change the authn method and initiate feature migration
+        with a rolling restart. Do not recreate tls certs.
+        '''
+        self.security.endpoint_authn_method = authn_method
+        self.redpanda.set_security_settings(self.security, init_tls=False)
```

Comment:
`init_tls` is set to False here because the certs were already created in the call to `prepare_cluster`, right?

Reply:
Correct.

Reply:
I'll add a comment and change the flag name to make that more clearn.

Reply:
Done.